p2p/peer/peerstore.go

```diff
@@ -38,20 +44,34 @@ type Peerstore interface {
 
 // AddressBook tracks the addresses of Peers
 type AddressBook interface {
-	Addresses(ID) []ma.Multiaddr
-	AddAddress(ID, ma.Multiaddr)
-	AddAddresses(ID, []ma.Multiaddr)
+	Addresses(ID) []ma.Multiaddr     // returns addresses for ID
+	AddAddress(ID, ma.Multiaddr)     // Adds given addr for ID
+	AddAddresses(ID, []ma.Multiaddr) // Adds given addrs for ID
+	SetAddresses(ID, []ma.Multiaddr) // Sets given addrs for ID (clears previously stored)
+}
+
+type expiringAddr struct {
+	Addr ma.Multiaddr
+	TTL  time.Time
+}
+
+func (e *expiringAddr) Expired() bool {
+	return time.Now().After(e.TTL)
 }
 
-type addressMap map[string]ma.Multiaddr
+type addressMap map[string]expiringAddr
 
 type addressbook struct {
 	addrs map[ID]addressMap
+	ttl   time.Duration // initial ttl
 	sync.RWMutex
```

Comment:
sure, i'm down to use that convention. i think there should be a comment declaring what it guards:

```go
type foo struct {
  amu // guards a
  a

  bmu // guards b
  b
}
```